alacritty_terminal/src/ansi.rs

```diff
@@ -991,6 +997,13 @@ where
                 handler.clear_line(mode);
             },
             ('S', None) => handler.scroll_up(Line(arg_or_default!(idx: 0, default: 1) as usize)),
+            ('t', None) => match arg_or_default!(idx: 0, default: 1) as usize {
+                22 => handler.push_title(),
+                23 => handler.pop_title(),
+                _ => {
+                    unhandled!();
+                },
```

Comment:
We don't need a dedicated block for this, it can just be written on a single line.

Reply:
Got it. One problem is that this function `csi_dispatch` has a cognitive complexity of 26 vs. the recommended 24. Should we ignore the warning with the function attribute `#[allow(clippy::cognitive_complexity)]` or refactor it?

Reply:
I don't see how this would change anything about that? I'm just suggesting to change the formatting.

Reply:
Oh yeah. I'm not suggesting that. With or without the change I get a Clippy warning.

Reply:
We should try to resolve cognitive complexity errors rather than ignoring them. I can take a look if you're having trouble figuring something out. Though this has been an issue with this function multiple times before.

Reply:
I'll start looking and see what I can do. Any advice is appreciated, of course!

Reply:
It's been interesting trying to understand cognitive complexity...I've never come across this Clippy lint until now.

One thing I found would be possible, is to combine the `'l'` and `'t'` arms. It lowers the cognitive complexity, but perhaps it's a little contrived. What do you think?

```rust
(action @ 'l', intermediate) | (action @ 'h', intermediate) => {
    let is_private_mode = match intermediate {
        Some(b'?') => true,
        None => false,
        _ => {
            unhandled!();
            return;
        },
    };
    for arg in args {
        let mode = Mode::from_primitive(is_private_mode, *arg);
        match (action, mode) {
            ('l', Some(mode)) => handler.unset_mode(mode),
            ('h', Some(mode)) => handler.set_mode(mode),
            _ => {
                unhandled!();
                return;
            },
        }
    }
},
```

Reply:
I don't get any cognitive complexity warning in your latest revision, is this still relevant?